lesson01/assignment/scripts.js

```diff
@@ -29,5 +29,120 @@ document. addEventListener("DOMContentLoaded", function(){
     // Clear the form field so the user can try again
     inputField.value = ''
   })
+})
+
+document. addEventListener("DOMContentLoaded", function(){
+
+  // Find the example form, and handle its submit event
+  const form1 = document.querySelector('#form1')
+
+  form1.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault()
+
+    // An array that we check based on user input
+    const people = [
+    {name: 'Victor', age: 30},
+    {name: 'Gail', age: 28},
+    {name: 'John', age: 31},
+    {name: 'Nhi', age: 60},
+    {name: 'Daniel', age: 25},
+    {name: 'Du', age: 70}
+    ];
+
+    const inputField = document.querySelector('#form1 .input'); //take the users information entered
+    const userInput = inputField.value; //converting user input into a constant called userInput
+    const person = people.find(({name}) => name === userInput); //look in the array and see if there is a name object that corresponds with the users input
+ 
+    // output the array item quantity based on user input
+    document.querySelector('#form1 .output').innerHTML = person.age; //return the age of the person found
+    document.querySelector('#form1 .name').innerHTML = userInput; //reeturn the name of the person found
+ 
+    // Clear the form field so the user can try again
+    inputField.value = ''
+  })
+})
+
+document. addEventListener("DOMContentLoaded", function(){
+
+  // Find the example form, and handle its submit event
+  const form2 = document.querySelector('#form2')
+
+  form2.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault()
+
+    // An array that we check based on user input
+    const cpuParts = ['CPU', 'RAM', 'GPU', 'PSU', 'motherboard', 'case', 'SSD', 'wireless card', 'fans', 'OS', 'operating system', 'hard drive']
+
+    const inputField = document.querySelector('#form2 .input'); //save the users data input as a constant called inputField
+    const userInput = inputField.value; //saving inputField to userInput
+ 
+    // output the array item quantity based on user input
+    document.querySelector('#form2 .output').innerHTML = cpuParts.includes(userInput); //return true if something typed is included from the array.
+    // Clear the form field so the user can try again
+    inputField.value = ''
+  })
+})
+
+document. addEventListener("DOMContentLoaded", function(){
+
+  // Find the example form, and handle its submit event
+  const form3 = document.querySelector('#form3')
+
+  form3.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault()
+
+    // An array that we can transform based on user input
+    const people = [
+    {name: 'Victor', age: 30},
+    {name: 'Gail', age: 28},
+    {name: 'John', age: 31},
+    {name: 'Nhi', age: 60},
+    {name: 'Daniel', age: 25},
+    {name: 'Du', age: 70}
+    ];
```

Comment:
Since this array is used by multiple functions, it could be declared just once at the top of the file and used as a global variable.